skyportal/handlers/api/comment.py

```diff
@@ -64,6 +64,15 @@ def post(self):
                         description: base64-encoded file contents
                       name:
                         type: string
+                  is_auto:
+                    type: boolean
+                    description: |
+                        specify True when a robot makes the comment.
+                        Defaults to False for human authors.
+                  altdata:
+                    type: object
+                    description: machine readable (searchable) data in JSON format.
+
```

Comment:
`altdata` and `text` creep into each other here somewhat. The content of `text` influences the content of `altdata`, and the content of `altdata` is handled different depending on whether `is_auto` is flagged. I think this is a sign that a separate database model is needed for autoannotations.  To the extent possible, I would like to keep our API for individual database classes as close to basic CRUD operations (https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) as possible. This makes the API contract presented to the user extremely clear, improves the long-term maintainability of the code, and makes it easier to extend the data model later on.